web/components/admin/TextField.tsx

```diff
@@ -128,25 +150,66 @@ export const TextField: FC<TextFieldProps> = ({
         </div>
       ) : null}
 
-      <div className="input-side">
-        <div className="input-group">
-          <Field
-            id={fieldId}
-            className={`field ${className} ${fieldId}`}
-            {...fieldProps}
-            {...(type !== TEXTFIELD_TYPE_NUMBER && { allowClear: true })}
-            placeholder={placeholder}
-            maxLength={maxLength}
-            onChange={handleChange}
-            onBlur={handleBlur}
-            onPressEnter={handlePressEnter}
-            disabled={disabled}
-            value={value as number | (readonly string[] & number)}
-          />
+      {!isAdminPwdField ? (
+        <div className="input-side">
+          <div className="input-group">
+            <Field
+              id={fieldId}
+              className={`field ${className} ${fieldId}`}
+              {...fieldProps}
+              {...(type !== TEXTFIELD_TYPE_NUMBER && { allowClear: true })}
+              placeholder={placeholder}
+              maxLength={maxLength}
+              onChange={handleChange}
+              onBlur={handleBlur}
+              onPressEnter={handlePressEnter}
+              disabled={disabled}
+              value={value as number | (readonly string[] & number)}
+            />
+          </div>
+          <FormStatusIndicator status={status} />
+          <p className="field-tip">{tip}</p>
+        </div>
+      ) : (
+        <div className="input-side">
+          <div className="input-group">
+            <Form
+              name="basic"
+              form={form}
+              initialValues={{ inputFieldPassword: value }}
+              style={{ width: '100%' }}
+            >
+              <Form.Item name="inputFieldPassword" rules={passwordComplexityRules}>
+                <Input.Password
+                  id={fieldId}
+                  className={`field ${className} ${fieldId}`}
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                  placeholder={placeholder}
+                  onPressEnter={handlePressEnter}
+                  disabled={disabled}
+                  value={value as number | (readonly string[] & number)}
+                />
+              </Form.Item>
+              {showPwdButton && (
```

Comment:
I see you added a submit button to `TextField`, but there's already a `TextFieldWithSubmit` that has a built-in submit button. It seems like these are now somewhat the same thing with them both showing a submit button on change. Sorry if I'm a little confused about needing to add a button to the one without a button.

Reply:
Hi, @gabek! The main reason why I add the separate submit button for the Admin Password field is that if we use the generic submit button of `TextFieldWithSubmit`, then the user would be able to submit the input even if their input is not satisfying the password complexity rules. Here in this post, I attached the screenshot example for this case. The separate submit button of `TextField` essentially prevents that problem and enforces the user to type a password that satisfies the password complexity rules. 

The screenshot example:
<img width="708" alt="Screen Shot 2023-02-21 at 3 22 51 PM" src="https://user-images.githubusercontent.com/26856199/220371875-60858fde-2f2b-4996-bcb5-63daa47c7ed0.png">